src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1302,22 +1302,23 @@ protected unsafe void WriteEventWithRelatedActivityIdCore(int eventId, Guid* rel
                 Debug.Assert(m_eventData != null);  // You must have initialized this if you enabled the source.
                 try
                 {
-                    EventOpcode opcode = (EventOpcode)m_eventData[eventId].Descriptor.Opcode;
-                    EventActivityOptions activityOptions = m_eventData[eventId].ActivityOptions;
+                    ref EventMetadata metadata = ref m_eventData[eventId];
```

Comment:
EventMetadata is a class. I would be more efficient to retrieve the reference to the instance once.

```suggestion
                    EventMetadata metadata = m_eventData[eventId];
```

Reply:
It's a struct